src/components/location/Location.js

```diff
@@ -0,0 +1,84 @@
+import './location.scss';
+import $ from 'jquery';
+
+class GoogleMap {
+  constructor(container, position) {
+    this.container = container;
+	  this.position = position;
+    this._googleMapKey = 'AIzaSyBApIxlKjCG4nPYE2GnZa6UltnP8JqMdgA';
+    this.zoom = 15;
+    this.checkPosition();
+  }
+
+  checkPosition() {
+    if(this.position instanceof Function) {
+      this.position().then((pos) => {
+        this.position = pos;
```

Comment:
it is bad practice to redefine some entity (ie if it is set as a function, it should be a function during the whole life of class)

Reply:
I've made it in order to have possibility to set coordinates both as object, or as callback function
In my example I use just callback, but it's possibly to use an object

Reply:
I see what you have done and it is a bad practice, it is not clear what's going on here. Why don't create `getPosition` method and `position` property?